src/renderer/components/reader/Reader.tsx

```diff
@@ -145,19 +138,22 @@ const computeReadiumCssJsonMessage = (): IEventPayload_R2_EVENT_READIUMCSS => {
     const jsonMsg: IEventPayload_R2_EVENT_READIUMCSS = { setCSS: cssJson };
     return jsonMsg;
 };
-setReadiumCssJsonGetter(computeReadiumCssJsonMessage);
 
-const publicationJsonUrl = queryParams.pub.startsWith(READIUM2_ELECTRON_HTTP_PROTOCOL) ?
-    convertCustomSchemeToHttpUrl(queryParams.pub) : queryParams.pub;
+const queryParams = getURLQueryParams();
+const lcpHint = queryParams.lcpHint;
+// pub is undefined when loaded in dependency injection by library webview.
+// Dependency injection is shared between all the renderer view
+const publicationJsonUrl = queryParams.pub?.startsWith(READIUM2_ELECTRON_HTTP_PROTOCOL)
```

Comment:
Is the question mark `?` and the code comment necessary here? If I am not mistaken, the URL query parameter `pub` is available immediately and reliably in the `BrowserWindow` / `webContents` URL:

https://github.com/readium/readium-desktop/blob/aacfe6bbd33db5623a01cfb2939713b6015c8790/src/main/redux/sagas/reader.ts#L130-L149

Reply:
no, it's always necessary here, the Reader code is imported in DI. And the DI is shared between reader and library. I am working to improve that in the future PR

Reply:
Ah yes of course, WebPack's statically-created bundle (at build time) imports the entire code hierarchy  ... so we need a separate PR to split the reader / library scopes.